var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -11,13 +12,15 @@ class Netcdf(Package):
     version('4.4.0', 'f01cb26a0126dd9a6224e76472d25f6c')
     version('4.3.3', '5fbd0e108a54bd82cb5702a73f56d2ae')
 
+    variant('mpi', default=True, description='Enables MPI parallelism')
     variant('fortran', default=False, description="Download and install NetCDF-Fortran")
     variant('hdf4',    default=False, description="Enable HDF4 support")
 
     # Dependencies:
     depends_on("curl")  # required for DAP support
     depends_on("hdf", when='+hdf4')
-    depends_on("hdf5")  # required for NetCDF-4 support
+    depends_on("hdf5+mpi~cxx", when='+mpi')  # required for NetCDF-4 support
```

Comment:
This has nothing to do with NetCDF-4. If you take a look at #400, you'll see that I added this comment to explain that HDF5 and zlib are required if you want NetCDF-4 support. From `./configure -h` for NetCDF 4.4.0:
```
  --disable-netcdf-4      do not build with netcdf-4 (else HDF5 and zlib required)
```
So HDF5 and zlib are really only dependencies if you want netcdf-4, which we do.

If you check the comments in the HDF5 package, you'll see the reason for `~cxx`:
```
The HDF5 configure script warns if cxx and mpi are enabled together. There doesn't seem to be a real reason for this, except that parts of the MPI interface are not accessible via the C++ interface. Since they are still accessible via the C interface, this is not actually a problem.
```
So it sounds like you should be able to remove the `~cxx` if you want to. But yeah, it's best to comment it better. How about something like:
```python
depends_on('m4')
depends_on('hdf', when='+hdf4')

# Required for DAP support
depends_on('curl')

# Required for NetCDF-4 support
depends_on('zlib')
depends_on('hdf5~mpi', when='~mpi')
depends_on('hdf5+mpi~cxx', when='+mpi') # mpi and cxx aren't compatible
```